Chat/Network.cpp

```diff
@@ -0,0 +1,183 @@
+#include "Network.h"
+#include <iostream>
+
+Network::Network() : is_working_(false){}
+
+Network::~Network()
+{
+    Cleanup();
+}
+
+int Network::PrepareNetwork()
+{
+    WSAData wsa_data;
+    int error_check = WSAStartup(WSA_VERSION, &wsa_data);
+    if(error_check != 0)
+    {
+        return error_check;
+    }
+
+    error_check = broadc_socket_.Initialize();
+    if(error_check != 0)
+    {
+        return error_check;
+    }
+
+    error_check = recv_socket_.Initialize();
+    if(error_check != 0)
+    {
+        return error_check;
+    }
+
+    //getting local pc ip
+    int rand_value = rand(), buffer = 0;
+    void *rand_value_send = NULL;
+    buffer = Parcer::PackMessage(PREPARE_MESSAGE, &rand_value, rand_value_send);
+    error_check = broadc_socket_.Send(rand_value_send, buffer);
+    if(error_check == SOCKET_ERROR)
+    {
+        return error_check;
+    }
+    
+    while(true)
+    {
+        RecvStruct recv_struct;
+        recv_socket_.Recv(&recv_struct);
+
+        UnpackedMessage unp_msg = Parcer::UnpackMessage(recv_struct.packet_);
+        if(unp_msg.type_ == PREPARE_MESSAGE ||
+           *(int*)unp_msg.msg_ == rand_value)
+        {
+            my_ip_ = recv_struct.ip_;
+            delete[] unp_msg.msg_;
+            break;
+        }
+
+        delete[] unp_msg.msg_;
+    }
+    //-----------------------
+
+    return 0;
+}
+
+int Network::SendMsg(UserMsg user_msg)
+{
+    void *packet = NULL;
+    int packet_size = Parcer::PackMessage(CHAT_MESSAGE, &user_msg, packet); //allocation in heap
+
+    send_mutex_.lock();
+    packet_size = broadc_socket_.Send(packet, packet_size);
+    send_mutex_.unlock();
+
+    delete[] packet; //clearing heap
+    return packet_size;
+}
+
+void Network::Cleanup()
```

Comment:
This function is called in destructor. It is good. But I saw pointer field in this class. I'm not sure it is free here.